src/SerializerGenerator.php

```diff
@@ -254,4 +254,13 @@ private function generateCodeForArray(
 
         return $this->templating->renderLoopArray($arrayPath, $modelPath, $index, $innerCode);
     }
+
+    private static function isArrayForPrimitive(PropertyTypeArray $type): bool
+    {
+        return match ($type->getSubType()::class) {
+            PropertyTypePrimitive::class => true,
+            PropertyTypeArray::class => self::isArrayForPrimitive($type->getSubType()),
+            default => false,
+        };
```

Comment:
i would prefer to do this without recursion. while it normally should work, i am sure somebody does some weird nesting where this becomes problematic. avoiding the recursion is also more efficient.

```
do {
    $type = $type->getSubType();
    if ($type instanceof PropertyTypePrimitive) {
        return true;
    }
while ($type instanceof PropertyTypeArray);

return false;
```